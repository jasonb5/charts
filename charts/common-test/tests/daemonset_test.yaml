suite: test daemonset
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: daemonset

      minReadySeconds: 10
      revisionHistoryLimit: 20
      updateStrategy:
        type: RollingUpdate
        rollingUpdate:
          maxSurge: 2
          maxUnavailable: 2
    asserts:
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should pass
    set:
      workload: daemonset
    asserts:
      - matchSnapshot: {}
        documentIndex: 0
      - hasDocuments:
          count: 2
