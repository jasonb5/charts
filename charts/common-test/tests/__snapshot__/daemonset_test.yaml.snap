should overwrite defaults:
  1: |
    minReadySeconds: 10
    revisionHistoryLimit: 20
    selector:
      matchlabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
          common/workload: default
      spec:
        containers:
          - image: busybox:0.1.0
            name: common-test
    updateStrategy:
      rollingUpdate:
        maxSurge: 2
        maxUnavailable: 2
      type: RollingUpdate
should pass:
  1: |
    apiVersion: app/v1
    kind: DaemonSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      selector:
        matchlabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
          common/workload: default
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: common-test
            common/workload: default
        spec:
          containers:
            - image: busybox:0.1.0
              name: common-test
      updateStrategy:
        type: OnDelete
